mapflow/functional/layer_utils.py

```diff
@@ -229,6 +230,14 @@ def generate_vector_layer(layer_uri,
     )
     return layer
 
+def footprint_to_extent(footprint):
```

Comment:
Please, use type annotations and return type hints. 
It is especially important in the functional module as it is used outside of the module and we need to keep track without documentation.

Also the docstrings may be of some use here, but in many cases types + function name are enough to understand what it is doing.


```def footprint_to_extent(footprint: dict) -> QgsRectangle:```
(check the types I used here! I did not check them)

Also, as th type of footprint is just a plain dict and it does not say anything particular about the input data expected, you may do one of two:
- (easy) add docstrings, specify that it is a `geojson-like dict (geometry)`
- best, but let's skip it by now: add a special class to represent the geojson-like geometry and use this class here (and in ImageReturnSchema as well)